cobra/io/mat.py

```diff
@@ -25,18 +25,18 @@
 
 
 # precompiled regular expressions
-_bracket_re = re.compile(r"\[[a-z]\]$")
-_underscore_re = re.compile(r"_[a-z]$")
+_bracket_re = re.compile(r"\[[a-z]{1,3}\]$")
+_underscore_re = re.compile(r"_[a-z]{1,3}$")
 
 
 def _get_id_compartment(id):
     """extract the compartment from the id string"""
     bracket_search = _bracket_re.findall(id)
     if len(bracket_search) == 1:
-        return bracket_search[0][1]
+        return bracket_search[0][1:]
```

Comment:
For a single letter, this was extracting just the compartment letter. So to achieve the same with your new expression, you will have to ignore the closing square bracket.

```suggestion
        return bracket_search[0][1:-1]
```

Reply:
Dear Moritz, 
yes I agree, I overlocked this one I was only testing my changes for a models with underscore notation.
I will test your suggestions for models with the bracket notation. 
For the test it is not necessary to have Matlab licenses, an exported model with the respective notation is sufficient. 

Best wishes,
Daniel